src/core/traits/scalar.rs

```diff
@@ -385,6 +385,6 @@ fn test_scalar_acos() {
     assert_approx_eq!((-2.0_f32).acos_approx(), core::f32::consts::PI);
 
     // input is clamped to -1.0..1.0
-    assert_eq!(2.0_f64.acos_approx(), 0.0);
-    assert_eq!((-2.0_f64).acos_approx(), core::f64::consts::PI);
+    assert!((2.0_f64.acos_approx() - 0.0).abs() < f64::EPSILON);
```

Comment:
It's been a while since I wrote this but I think in this case I do want to check that the value is exactly equal, not within epsilon.

Reply:
It seem clippy doesn't detect the case of being equal to zero so i will just revert that one and put a allow to silence it.

Reply:
https://github.com/rust-lang/rust-clippy/issues/3804